### PR TITLE
validation done

### PR DESCRIPTION
**Pull request by peter-wilkins**
HI Mike

Here's my challenge.

### CODE REVIEW

File: js/validator.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                'Required if above radio is Yes' +
+                '</div>');
+            console.log('radio is yes and field6 is empty');
+
+        }
+        else {
+            $("#err6").alert('close');
+        }
+        //Valid email
+        var testEmail = /[A-Z0-9._%+-]+@[A-Z0-9.-]+.[A-Z]{2,4}/igm;
```
Comment (mporam) [inline js/validator.js]: you havent escaped your . in the regex so it reads it as the regex character . which means anything, this results in you being able to leave the .com off the email address, or put .kfjgsjrhbgvfhbvsz and it still being valid

File: js/validator.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                    '</div>');
+                console.log('field4 wrong');
+            }
+            else {
+                $("#err4").alert('close');
+            }
+        }
+
+        //Letters only
+        var testLetters = /^[a-zA-Z]+$/;
```
Comment (mporam) [inline js/validator.js]: this doesnt allow spaces. I no the validation is ambiguous about this but how would you make it allow spaces if you wanted to?

File: js/validator.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        // required
+        if (!$('#field1').val()) {
+            $("#err1").alert('close');
+            $('#field1').after(
+                '<div class="alert alert-danger alert-dismissable" id="err1">' +
+                '<button type="button" class="close" ' +
+                'data-dismiss="alert" aria-hidden="true">' +
+                '&times;' +
+                '</button>' +
+                'Required field' +
```
Comment (mporam) [inline js/validator.js]: the validation message is exactly the same as the field label, maybe you could grab the text from there to save typing it in?

**Comment by mporam** [comment]
this looks really good Pete, iv added some comments of small mistakes in your regex. Only other thing I would have liked is your error messages do not change depending on what the user has input. for example if the user has put 26 characters in a max 25 character input the message only needs to say maximum 25 characters rather than the whole validation message. maybe something to improve on if you get chance.

**Comment by peter-wilkins-mayden** [comment]
Implemented your suggestions + Mayden standards

**Comment by mporam** [comment]
This looks really good now pete. Just to be supoer annoying, the required field with a min max length, If you leave it blank you just get the require field message, it would be better to get that + the min length message.

Finally only other change I might make would be to abstract the error message code intto a function where you pass it a string and it creates the HTML for you as you have a lot of duplicate code for the alerts.

**Pull request closed without merging**